pandas/tests/indexes/multi/test_set_ops.py

```diff
@@ -85,63 +64,37 @@ def test_difference_base(idx):
     answer = idx[4:]
     result = first.difference(second)
 
-    if isinstance(idx, CategoricalIndex):
-        pass
-    else:
-        assert tm.equalContents(result, answer)
+    assert tm.equalContents(result, answer)
 
     # GH 10149
     cases = [klass(second.values)
              for klass in [np.array, Series, list]]
     for case in cases:
-        if isinstance(idx, PeriodIndex):
-            msg = "can only call with other PeriodIndex-ed objects"
-            with tm.assert_raises_regex(ValueError, msg):
-                result = first.difference(case)
-        elif isinstance(idx, CategoricalIndex):
-            pass
-        elif isinstance(idx, (DatetimeIndex, TimedeltaIndex)):
-            assert result.__class__ == answer.__class__
-            tm.assert_numpy_array_equal(result.sort_values().asi8,
-                                        answer.sort_values().asi8)
-        else:
-            result = first.difference(case)
-            assert tm.equalContents(result, answer)
-
-    if isinstance(idx, MultiIndex):
-        msg = "other must be a MultiIndex or a list of tuples"
-        with tm.assert_raises_regex(TypeError, msg):
-            result = first.difference([1, 2, 3])
+        result = first.difference(case)
+        assert tm.equalContents(result, answer)
+
+    msg = "other must be a MultiIndex or a list of tuples"
+    with tm.assert_raises_regex(TypeError, msg):
+        result = first.difference([1, 2, 3])
 
 
 def test_symmetric_difference(idx):
     first = idx[1:]
     second = idx[:-1]
-    if isinstance(idx, CategoricalIndex):
-        pass
-    else:
-        answer = idx[[0, -1]]
-        result = first.symmetric_difference(second)
-        assert tm.equalContents(result, answer)
+    answer = idx[[0, -1]]
+    result = first.symmetric_difference(second)
+    assert tm.equalContents(result, answer)
 
     # GH 10149
     cases = [klass(second.values)
```

Comment:
if you want to come back and parameterize things like this would be great (but future PR)